media-proxy/src/mesh/st2110_30rx.cc

```diff
@@ -0,0 +1,85 @@
+#include "st2110rx.h"
+
+namespace mesh::connection {
+
+ST2110_30Rx::ST2110_30Rx() {}
+
+ST2110_30Rx::~ST2110_30Rx() {}
+
+st30_frame *ST2110_30Rx::get_frame(st30p_rx_handle h) { return st30p_rx_get_frame(h); };
+
+int ST2110_30Rx::put_frame(st30p_rx_handle h, st30_frame *f) { return st30p_rx_put_frame(h, f); };
+
+st30p_rx_handle ST2110_30Rx::create_session(mtl_handle h, st30p_rx_ops *o)
+{
+    return st30p_rx_create(h, o);
+};
+
+int ST2110_30Rx::close_session(st30p_rx_handle h) { return st30p_rx_free(h); };
+
+Result ST2110_30Rx::configure(context::Context &ctx, const std::string &dev_port,
+                              const MeshConfig_ST2110 &cfg_st2110,
+                              const MeshConfig_Audio &cfg_audio)
+{
+    static int session_id;
+    static std::mutex mtx;
+    std::lock_guard<std::mutex> lock(mtx);
+
+    if (cfg_st2110.transport != MESH_CONN_TRANSPORT_ST2110_30) {
+        set_state(ctx, State::not_configured);
+        return set_result(Result::error_bad_argument);
+    }
+
+    _st = get_mtl_handle(dev_port, MTL_LOG_LEVEL_CRIT, cfg_st2110.local_ip_addr);
+    if (_st == nullptr) {
+        log::error("Failed to get MTL device");
+        set_state(ctx, State::not_configured);
+        return set_result(Result::error_bad_argument);
+    }
+
+    char session_name[NAME_MAX] = "";
+
+    snprintf(session_name, NAME_MAX, "mcm_rx_st30_%d", session_id++);
+
+    inet_pton(AF_INET, cfg_st2110.remote_ip_addr, _ops.port.ip_addr[MTL_PORT_P]);
+    inet_pton(AF_INET, cfg_st2110.local_ip_addr, _ops.port.mcast_sip_addr[MTL_PORT_P]);
+    _ops.port.udp_port[MTL_PORT_P] = cfg_st2110.local_port;
+
+    strlcpy(_ops.port.port[MTL_PORT_P], dev_port.c_str(), MTL_PORT_MAX_LEN);
+    _ops.port.num_port = 1;
+    _ops.port.payload_type = ST_APP_PAYLOAD_TYPE_ST30;
+    _ops.name = strdup(session_name);
```

Comment:
Can you confirm you delete the newly created string after it is not needed?

Reply:
No, you never delete it. Rework this to allow deleting it appropriately.

Reply:
Fixed in 2479ca6f6d060db56009026a4c727be98472c6dc